pkg/confmap/provider/signozenvprovider/provider.go

```diff
@@ -0,0 +1,94 @@
+package signozenvprovider
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	schemeName                string = "signozenv"
+	envPrefix                 string = "signoz"
+	separator                 string = "__"
+	envPrefixWithOneSeparator string = "signoz_"
+	envRegexString            string = `^[a-zA-Z][a-zA-Z0-9_]*$`
+)
+
+var (
+	envRegex = regexp.MustCompile(envRegexString)
+)
+
+type provider struct {
+	logger *zap.Logger
+}
+
+// NewFactory returns a factory for a confmap.Provider that reads the configuration from the environment.
+// All variables starting with `SIGNOZ__` are read from the environment.
+// The separator is `__` (2 underscores) in order to incorporate env variables having keys with a single `_`
+func NewFactory() confmap.ProviderFactory {
+	return confmap.NewProviderFactory(newProvider)
+}
+
+func newProvider(settings confmap.ProviderSettings) confmap.Provider {
+	return &provider{
+		logger: settings.Logger,
+	}
+}
+
+func (provider *provider) Retrieve(_ context.Context, uri string, _ confmap.WatcherFunc) (*confmap.Retrieved, error) {
```

Comment:
Why does it only support env? I think it is easier to use and manage with the file provider. Not everything can be configured with env, for instance, how can I configure different readers for the meter provider?